packages/@sanity/default-layout/src/navbar/components/collapseMenu/CollapseMenu.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {
+  useState,
+  useEffect,
+  useCallback,
+  useMemo,
+  ReactElement,
+  Children,
+  cloneElement,
+} from 'react'
+import {
+  Box,
+  Button,
+  Flex,
+  Menu,
+  MenuButton,
+  MenuItem,
+  Text,
+  Tooltip,
+  useElementRect,
+  ThemeColorSchemeKey,
+} from '@sanity/ui'
+import styled from 'styled-components'
+import {InView} from 'react-intersection-observer'
+import {EllipsisVerticalIcon} from '@sanity/icons'
+import {CollapseMenuDivider, CollapseMenuButton} from '.'
+
+interface CollapseMenuProps {
+  children: React.ReactNode
+  gap?: number | number[]
+  menuButton?: ReactElement<HTMLButtonElement>
+  menuScheme?: ThemeColorSchemeKey
+  onMenuVisible?: (visible: boolean) => void
+}
+
+const Root = styled(Box)<{$hide?: boolean}>`
+  border-radius: inherit;
+  overflow: hidden;
+  position: relative;
+  white-space: nowrap;
+  width: 100%;
+  padding: 0.25rem;
+  margin: -0.25rem;
+`
+
+const Inner = styled(Flex)<{$hide?: boolean}>`
+  inset: 0;
+  pointer-events: ${({$hide}) => ($hide ? 'none' : 'inherit')};
+  opacity: ${({$hide}) => ($hide ? 0 : 1)};
+  position: ${({$hide}) => ($hide ? 'absolute' : 'static')};
+  visibility: ${({$hide}) => ($hide ? 'hidden' : 'visible')};
+  width: ${({$hide}) => ($hide ? 'max-content' : 'auto')};
+`
+
+const OptionBox = styled(Box)`
+  display: flex;
+  flex-shrink: 0;
+  list-style: none;
+  white-space: nowrap;
```

Comment:
On smaller components I don't really mind, but it might be best in cases like this component to move these into their own file? :)